htsget_server/variants.py

```diff
@@ -0,0 +1,326 @@
+        ["H2", "0", "Flag", "HapMap2 membership"],
+        ["H3", "0", "Flag", "HapMap3 membership"],
+        ["MQ", "1", "Float", "RMS mapping quality"],
+        ["MQ0", "1", "Integer", "Number of MAPQ == 0 reads"],
+        ["NS", "1", "Integer", "Number of samples with data"],
+        ["SB", "4", "Integer", "Strand bias"],
+        ["SOMATIC", "0", "Flag", "Somatic mutation (for cancer genomics)"],
+        ["VALIDATED", "0", "Flag", "Validated by follow-up experiment"],
+        ["1000G", "0", "Flag", "1000 Genomes membership"]
+    ]
+    for r in reserved:
+        info_headers[r[0]] = {
+            "Number": r[1],
+            "Type": r[2],
+            "Description": r[3]
+        }
+    info_pieces = text.split(';')
+    info_obj = {}
+    for info in info_pieces:
+        kv = info.split('=', 2)
+        if kv[0] in info_headers:
+            info_obj[kv[0]] = {
+                'Type': info_headers[kv[0]]['Type'],
+                'Number': info_headers[kv[0]]['Number'],
+                'Description': info_headers[kv[0]]['Description'],
+                'Value': None
+            }
+            if len(kv) > 1:
+                vals = kv[1].split(',')
+                if info_obj[kv[0]]['Number'] == '1':
+                    info_obj[kv[0]]['Value'] = [kv[1]]
+                else:
+                    info_obj[kv[0]]['Value'] = vals
+
+    # find a CSQ header, as a special case:
+    csq_header = None
+    if 'CSQ' in info_headers:
+        csq_header = info_headers['CSQ']['Description']
+    if 'CSQ' in text and csq_header is not None:
+        info_obj['CSQ']['Description'] = "Consequence annotations from Ensembl VEP."
+        info_obj['CSQ']['Value'] = parse_vep_annotation(info_obj['CSQ']['Value'], csq_header)
+        info_obj['CSQ']['Number'] = 'K' # obj keyed by allele
+
+    return info_obj
+
+
+def parse_vep_annotation(info, csq_header):
+    result = {}
+    csq_match = re.match(r".+Format: (.+)", csq_header)
+    csq_parts = csq_match.group(1).split('|')
+    for i in range(len(info)):
+        this_info = {}
+        info_pieces = info[i].split('|')
+        if len(info_pieces) <= len(csq_parts):
+            for j in range(len(info_pieces)):
+                if info_pieces[j] is not None and info_pieces[j] != '':
+                    this_info[csq_parts[j]] = info_pieces[j]
+        if this_info['Allele'] not in result:
+            result[this_info['Allele']] = []
+        result[this_info['Allele']].append(this_info)
+    return result
+
+
+def get_genotype_index(a, b):
+    # from VCF spec:
+    # genotype ordering: the index of the genotype “a/b”, where a ≤ b, is b(b + 1)/2 + a
+    if a > b:
+        c = a
+        a = b
+        b = c
+    return ((b * (b + 1)) / 2) + a
+
+
+def seq_match(a, b):
+    return len(set(expand_iupac(a)).intersection(set(expand_iupac(b)))) > 0
+
+
+def expand_iupac(base_str):
+    result = []
+    ambig_match = re.match(r"(^.*?)([RYSWKMBDHVN])(.*$)", base_str)
```

Comment:
While I can sort of read this, can we add a comment above this? Something like:
```suggestion
    # Find ambiguous nucleotide sequences (https://www.bioinformatics.org/sms/iupac.html) in the given sequence.
    ambig_match = re.match(r"(^.*?)([RYSWKMBDHVN])(.*$)", base_str)
```